### PR TITLE
Removed any unnecessary data being passed to backend in extensions

### PR DESCRIPTION
**Pull request by ziafazal**
To make sure we are passing only useful data to caliper/xAPI LRS this PR removes unnecessary data. 
In addition to security risk involved sometimes this data makes xAPI statements invalid i.e `code` being passed to video events.

### CODE REVIEW

**Comment by ziafazal** [comment]
@timmc-edx could you please review this one?

**Comment by ziafazal** [comment]
@sheralim012 could you please review it?

**Review by sheralim012 — APPROVED** [review]

**Comment by ziafazal** [comment]
Closing this in favour of PR #62. I requested @sheralim012 to create a new PR with commit of this PR and then I'll review to unblock merging. This is a workaround we found to by pass `At least 1 approving review is required by reviewers with write access` restriction and merge edly PRs into master.

**Pull request closed without merging**
